### PR TITLE
feat: Add search to ProjectsDropdown

### PR DESCRIPTION
**Pull request by Mitcheljager**
This PR adds a search box to the ProjectsDropdown component. This allows you to filter the dropdown contents by their title. The search component is fairly generic, allowing it to be used for any other array of objects for any key. The results can be retrieved via a two way bind.

## Filter
search-flip

## Empty
image

### CODE REVIEW

File: app/javascript/src/components/editor/ProjectsDropdown.svelte [hunk 1]
```diff
@@ -187,12 +192,22 @@
...
+      <hr />
+
+      {#each filteredProjects as project (project.uuid)}
+        <div class="dropdown__item" animate:flip={{ duration: 100 }} on:click={() => fetchProject(project.uuid)}>
           {project.title}
         </div>
       {/each}
 
+      {#if $projects?.length && !filteredProjects.length}
+        <em class="block text-dark text-small pl-1/8 pr-1/8">No projects match your filter.</em>
```
Comment (CactusPuppy) [inline app/javascript/src/components/editor/ProjectsDropdown.svelte:208]:
```suggestion
        <em class="block text-dark text-small pl-1/8 pr-1/8">No projects match your search.</em>
```

**Review by CactusPuppy — APPROVED** [review]
LGTM!

File: app/javascript/src/components/editor/SearchObjects.svelte [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+  let query = ""
+
+  $: if (objects) search(query)
+
+  function search() {
+    if (query) filteredObjects = objects.filter(o => o[key].toLowerCase().indexOf(query.toLowerCase()) !== -1)
+    else filteredObjects = objects
+
+  }
```
Comment (CactusPuppy) [inline app/javascript/src/components/editor/SearchObjects.svelte:15]: Was it really not possible to make this a single `$:` statement somehow?

Comment (Mitcheljager) [reply app/javascript/src/components/editor/SearchObjects.svelte:15]: It may have been! This feels a little neater though

**Pull request merged**